handler/handler.go

```diff
@@ -0,0 +1,133 @@
+package handler
+
+import (
+	"encoding/csv"
+	"io"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/ONSdigital/dp-dimension-extractor/dimension"
+	"github.com/ONSdigital/dp-dimension-extractor/instance"
+	"github.com/ONSdigital/dp-dimension-extractor/schema"
+	"github.com/ONSdigital/go-ns/kafka"
+	"github.com/ONSdigital/go-ns/log"
+	"github.com/ONSdigital/go-ns/s3"
+)
+
+type dimensionExtracted struct {
+	FileURL    string `avro:"file_url"`
+	InstanceID string `avro:"instance_id"`
+}
+
+type inputFileAvailable struct {
+	FileURL    string `avro:"file_url"`
+	InstanceID string `avro:"instance_id"`
+}
+
+// HandleMessage handles a message by sending requests to the import API
+// before producing a new message to confirm successful completion
+func HandleMessage(producer kafka.Producer, s3 *s3.S3, importAPIURL string, message kafka.Message, maxRetries int) (string, error) {
+	event, err := readMessage(message.GetData())
+	if err != nil {
+		log.Error(err, log.Data{"schema": "failed to unmarshal event"})
+		return "", err
+	}
+
+	log.Debug("event received", log.Data{"file_url": event.FileURL, "instance_id": event.InstanceID})
+
+	// Get csv from S3 bucket using m.S3URL
+	file, err := s3.Get(event.FileURL)
+	if err != nil {
+		log.ErrorC("encountered error retrieving csv file", err, log.Data{"instance_id": event.InstanceID})
+		return event.InstanceID, err
+	}
+
+	log.Debug("file successfully read from aws", log.Data{"instance_id": event.InstanceID})
+
```

Comment:
Also, this whole package I think should be moved into service. Service is passing in 4 arguments that are attributes of service, so hadler.HandleMessage(1,2,3,4,5) could become svc.Handle(message) and be potentially cleaner.

This would probably give even more options for separating this out into other methods, further to my previous suggestion.